website/blog/2020-05-04-lithium.md

```diff
@@ -0,0 +1,312 @@
+- automatically add '+' on newline inside a string
+- code action "Import all missing symbols"
+
+## Basic Scala 3 support
+
+For a couple of last months we have been working on making Metals work with the
+Dotty compiler, which will become the new Scala 3. This effort included
+contributions to Metals, Bloop as well as to Dotty. Thanks to a great help from
+the Dotty compiler team, we now have an option to generate semanticDB files used
+for our internal indexes, which allowed some features to work without any
+changes to the core code. It is also important to mention that without
+[jvican](https://github.com/jvican) and his Bloop project, which already
+supports Scala 3, all this would have taken a much longer time.
+
+That said, there are some issues that are lacking in the Scala 3 support that
+are available for 2.x versions. Most notably, we are still missing all of the
+more advanced completions, filtering and sorting of completion items or auto
+imports. There are also some possible issues connected to optional braces and
+top level methods features. In the upcoming months we are planning to address
+most of those issues and implement the lacking features.
+
+You can find missing features in the
+[feature requests repository](https://github.com/scalameta/metals-feature-requests/issues?q=is%3Aissue+is%3Aopen+label%3Ascala3)
+and currently known bugs in the
+[main Metals repo.](https://github.com/scalameta/metals/issues?q=is%3Aissue+is%3Aopen+label%3A%22Scala+3%22)
+
+The first supported versions will be `0.23.0` and `0.24.0-RC1`, which are the
+latest stable and latests release candidate respectably. We plan to always
+support such two versions until there is a more stable release cycle for
+Scala 3.
+
+Please do report any issues you encounter, especially connected to some new
+Scala 3 features, which will be an immense help to the team.
+
+## Improvements related to Bloop/Metals integration
+
+There has been multiple reported issues in the last weeks about hanging
+compilation, which could make the Metals workspace hard to use. Our top priority
+is currently figuring out all the possible issues both in Metals as well in
+Bloop. Thanks to some fixes done by [jvican](https://github.com/jvican) we are
+hopeful that some of the issues might become less frequent. Those fixes are
+available in the current Bloop snapshots and in the version included in Metals
+
+We also reworked the Metals tree view to highlight some commands that a user can
+use to fix their workspace as well as added a new command to recompile workspace
+from scratch:
+
+![reworked-tree-view](https://i.imgur.com/PERfNqt.png)
+
+`Clean compile workspace` might be needed in case of broken Zinc caches and
+`Restart Bloop server` should help with the cases where deduplication of
+compilation would hang. Do not hesitate to report any possible issues and if
+possible there are a number of additional diagnostics that can be done before
+reporting an issue:
```

Comment:
Tbh I know this is extremely helpful to have for an issue, but I'm feeling a bit like these release notes are starting to sway pretty heavy on the tech side rather than an overview of the features and functionality this release has to offer. Again, I totally think we should include some of the technical improvements, but I'm not sure how I feel about telling users to "Give us a thread dump of the Bloop process" here. Imagine a person new to Scala just getting started with Metals reading over these release notes? Thoughts?

Reply:
Maybe I went over the board :sweat_smile: . I will reduce it to "please report an issue". We can guide the user later. Though it would be extremely useful have that reported right from the bat, I spent loads of time on digging and didn't manage to get anywhere. However, I agree it might be too much.

Reply:
Changed the section a bit and now it's just:
`Even if those commands help, do not hesitate to report any issues. We are also
monitoring the situation ourselves and investigating all problems encountered.`

Reply:
Yea I agree it's important. Maybe we should re-work the issue template to say that if they are dealing with this specific type of issue, these are the things that would be helpful.

Reply:
Maybe it will not be necessary. I haven't noticed the issue since Bloop updated, so here's hoping! :crossed_fingers: